lib/jaeger/samplers/guaranteed_throughput_probabilistic.rb

```diff
@@ -30,16 +30,16 @@ def update(lower_bound:, rate:)
         is_updated
       end
 
-      def sample(*args)
-        is_sampled, probabilistic_tags = @probabilistic_sampler.sample(*args)
+      def sample(**args)
```

Comment:
I think this is the only real part that confuses me. Can we just go from * (destructured array) arguments to ** (destructured hash keyword arguments)?

Reply:
TLDR; There is little more than straight destructuring happening in the old code, and we are turning it into straight destructuring with the new code.

There are a couple funky things happening here, but at the root of it, many of the samplers delegate to `probabilistic#sample`, which contains a mix of a `trace_id` kwargs and an options `**`. However, those samplers are using the old delegation syntax (`*args`) which now destructures into positional arguments.

Ruby <= 2.6 uses `*` (like `*args`) to pass arguments to delegate methods. Ruby >= 2.7 uses a different style in an attempt to separate positional and keyword args. (there is even a new delegation syntax `...` — that actually might be the right thing to do here.

The meat of it is in the "Handling argument delegation" section of https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/

Reply:
And since the end of the delegation chain uses kwargs, we need to explicitly start with kwargs.

Reply:
This comment inspired me to take a less intrusive approach.